src/Roslyn.Suppressions/Roslyn.Suppressions.UnitTests/InProcess/SuppressedIssuesFileSynchronizerTests.cs

```diff
@@ -124,6 +126,43 @@ public void OnSuppressionsUpdateRequested_ConnectedMode_StorageUpdated(bool hasI
             suppressedIssuesFileStorage.Verify(x=> x.Update("some project key", issues), Times.Once);
         }
 
+        [TestMethod]
+        public async Task UpdateFileStorage_FileStorageIsUpdatedOnBackgroundThread()
+        {
+            Func<Task<bool>> fileUpdateTask = null;
+            var threadHandling = new Mock<IThreadHandling>();
+            threadHandling
+                .Setup(x => x.Run(It.IsAny<Func<Task<bool>>>()))
+                .Callback((Func<Task<bool>> task) => fileUpdateTask = task);
+
+            var configuration = CreateConnectedConfiguration("some project key");
+            var activeSolutionBoundTracker = new Mock<IActiveSolutionBoundTracker>();
```

Comment:
Minor: all of the tests would be a little simpler if there were helper methods to create the `activeSolutionBoundTracker` and `suppressedIssuesProvider`.